LBH.AdultSocialCare.Functions.Payruns/Services/InvoiceItemGenerators/FundedNursingCareGenerator.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LBH.AdultSocialCare.Api.Helpers;
+using LBH.AdultSocialCare.Functions.Payruns.Enums;
+using LBH.AdultSocialCare.Functions.Payruns.Gateways.Interfaces;
+using LBH.AdultSocialCare.Functions.Payruns.Infrastructure.Entities.CarePackages;
+using LBH.AdultSocialCare.Functions.Payruns.Infrastructure.Entities.Common;
+using LBH.AdultSocialCare.Functions.Payruns.Infrastructure.Entities.Payments;
+
+namespace LBH.AdultSocialCare.Functions.Payruns.Services.InvoiceItemGenerators
+{
+    public class FundedNursingCareGenerator : BaseInvoiceItemsGenerator
+    {
+        private List<FundedNursingCarePrice> _fncPrices;
+        private readonly IFundedNursingCareGateway _fundedNursingCareGateway;
+
+        public FundedNursingCareGenerator(IFundedNursingCareGateway fundedNursingCareGateway)
+        {
+            _fundedNursingCareGateway = fundedNursingCareGateway;
+        }
+
+        public override IEnumerable<InvoiceItem> Run(CarePackage package, DateTimeOffset invoiceStartDate, DateTimeOffset invoiceEndDate)
+        {
+            var invoiceItems = new List<InvoiceItem>();
+            var fundedNursingCare = package.Reclaims
+                .FirstOrDefault(r => r.Type is ReclaimType.Fnc);
+
+            if (fundedNursingCare is null) return invoiceItems;
+
+            foreach (var price in _fncPrices)
+            {
+                var actualStartDate = Dates.Max(price.ActiveFrom, invoiceStartDate);
+                var actualEndDate = Dates.Min(price.ActiveTo, invoiceEndDate);
```

Comment:
Only one fnc will be applied and that depends on customer age I believe

Reply:
FNC price is changing each year at April 1st, so if package covers different FNC price ranges, different items are generated

Was it changed somehow to depend on client's age?

Reply:
Hmm. Just looked at the table looks like the age range thing was removed. In that case this will add FNC for all years but believe the date changes are coming with date handling to only select active FNC. Okay

Reply:
There never was an age on FNC, so I believe there is confusion between provisional care charge which is age-based, and FNC, which is changed every year

Reply:
Yess. That's the one. My bad. I will merge now